### PR TITLE
add other toggle to both tables

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/12308

the pr is branched off https://github.com/cityofaustin/atd-vz-data/pull/1273 for easier readability

this pr adds 'other' mode toggles to the crashes and fatalities tables. if all data in vzd is correct, selecting all the toggles should yield the same number of records as selecting none.

if the all toggles state is less than the no toggles state, then there are likely crashes without units associated with them. i mentioned to xavier there is at least one crash without associated units (19530310) and he said he is aware of this and referenced [this issue](https://github.com/cityofaustin/atd-data-tech/issues/7622) as being unresolved and blocking this work.

## Testing
**URL to test:** 
local

**Steps to test:**
- open a local instance of vze using a current read replica
- open the fatalities table and set the range to 2014-07-28 to 2023-07-28
- toggle on and off the 'other' toggle to see only crashes associated with units that don't fit into the categories above it
- note the total number of records with no mode toggles on

Screenshot 2023-08-25 at 12 25 03 PM

- turn all the mode toggles on (including 'other') - the number should not change

Screenshot 2023-08-25 at 12 23 27 PM

- test out different date ranges

- repeat these steps with the crashes table
- you will see that the all toggles state is missing one crash as compared to the no toggles state

Screenshot 2023-08-25 at 12 26 25 PM
Screenshot 2023-08-25 at 12 26 51 PM

- duplicate any record in the units table with an associated crash and change the `crash_id` to `19530310`
- save your changes, open the crash record in vze for `19530310` and confirm the duplicated unit appears in the units table
- go back to the crashes table and turn on all the toggles

Screenshot 2023-08-25 at 2 26 45 PM

- the number should now be the same as without any toggles on

Screenshot 2023-08-25 at 12 26 51 PM

- we should talk as a team about how to address the issue that blocks the vz team from fixing this data discrepancy

if you want, you can run this query on vzd to see all the totals. please note you'll probably see that there are still uncaptured units.

```
WITH all_people AS
(
       SELECT person.prsn_nbr,
              person.crash_id,
              person.unit_nbr,
              person.prsn_injry_sev_id,
              'nonprimary' AS class
       FROM   atd_txdot_person person
       WHERE  person.prsn_injry_sev_id = 4
       UNION
       SELECT primaryperson.prsn_nbr,
              primaryperson.crash_id,
              primaryperson.unit_nbr,
              primaryperson.prsn_injry_sev_id,
              'primary' AS class
       FROM   atd_txdot_primaryperson primaryperson
       WHERE  primaryperson.prsn_injry_sev_id = 4)
SELECT *,
       (fatalities_total - ( fatalities_motor_vehicle + fatalities_motorcycle + fatalities_cyclist + fatalities_pedestrian + fatalities_scooter + fatalities_other )) AS uncaptured_fatalities,
       (units_total      - ( units_motor_vehicle + units_motorcycle + units_cyclist + units_ped + units_scooter + units_other ))                                      AS uncaptured_units
FROM   (
              SELECT
                     (
                               SELECT    count(DISTINCT(crashes.crash_id, units.unit_nbr, people.prsn_nbr)) AS fatalities_total
                               FROM      atd_txdot_crashes crashes
                               LEFT JOIN atd_txdot_units units
                               ON        (
                                                   crashes.crash_id = units.crash_id)
                               LEFT JOIN all_people people
                               ON        (
                                                   people.crash_id = crashes.crash_id
                                         AND       people.unit_nbr = units.unit_nbr)
                               WHERE     true
                               AND       crashes.crash_date >= '2014-07-28'
                               AND       crashes.crash_date <= '2023-07-28'
                               AND       people.prsn_injry_sev_id = 4
                               AND       crashes.in_austin_full_purpose IS true),
                     (
                               SELECT    count(DISTINCT(crashes.crash_id, units.unit_nbr, people.prsn_nbr)) AS fatalities_motor_vehicle
                               FROM      atd_txdot_crashes crashes
                               LEFT JOIN atd_txdot_units units
                               ON        (
                                                   crashes.crash_id = units.crash_id)
                               LEFT JOIN all_people people
                               ON        (
                                                   people.crash_id = crashes.crash_id
                                         AND       people.unit_nbr = units.unit_nbr)
                               WHERE     true
                               AND       crashes.crash_date >= '2014-07-28'
                               AND       crashes.crash_date <= '2023-07-28'
                               AND       units.unit_desc_id = 1
                               AND       units.veh_body_styl_id NOT IN (71,90)
                               AND       people.prsn_injry_sev_id = 4
                               AND       crashes.in_austin_full_purpose IS true),
                     (
                               SELECT    count(DISTINCT(crashes.crash_id, units.unit_nbr, people.prsn_nbr)) AS fatalities_motorcycle
                               FROM      atd_txdot_crashes crashes
                               LEFT JOIN atd_txdot_units units
                               ON        (
                                                   crashes.crash_id = units.crash_id)
                               LEFT JOIN all_people people
                               ON        (
                                                   people.crash_id = crashes.crash_id
                                         AND       people.unit_nbr = units.unit_nbr)
                               WHERE     true
                               AND       crashes.crash_date >= '2014-07-28'
                               AND       crashes.crash_date <= '2023-07-28'
                               AND       units.unit_desc_id = 1
                               AND       units.veh_body_styl_id IN (71,90)
                               AND       people.prsn_injry_sev_id = 4
                               AND       crashes.in_austin_full_purpose IS true),
                     (
                               SELECT    count(DISTINCT(crashes.crash_id, units.unit_nbr, people.prsn_nbr)) AS fatalities_cyclist
                               FROM      atd_txdot_crashes crashes
                               LEFT JOIN atd_txdot_units units
                               ON        (
                                                   crashes.crash_id = units.crash_id)
                               LEFT JOIN all_people people
                               ON        (
                                                   people.crash_id = crashes.crash_id
                                         AND       people.unit_nbr = units.unit_nbr)
                               WHERE     true
                               AND       crashes.crash_date >= '2014-07-28'
                               AND       crashes.crash_date <= '2023-07-28'
                               AND       units.unit_desc_id = 3
                               AND       people.prsn_injry_sev_id = 4
                               AND       crashes.in_austin_full_purpose IS true),
                     (
                               SELECT    count(DISTINCT(crashes.crash_id, units.unit_nbr, people.prsn_nbr)) AS fatalities_pedestrian
                               FROM      atd_txdot_crashes crashes
                               LEFT JOIN atd_txdot_units units
                               ON        (
                                                   crashes.crash_id = units.crash_id)
                               LEFT JOIN all_people people
                               ON        (
                                                   people.crash_id = crashes.crash_id
                                         AND       people.unit_nbr = units.unit_nbr)
                               WHERE     true
                               AND       crashes.crash_date >= '2014-07-28'
                               AND       crashes.crash_date <= '2023-07-28'
                               AND       units.unit_desc_id = 4
                               AND       people.prsn_injry_sev_id = 4
                               AND       crashes.in_austin_full_purpose IS true),
                     (
                               SELECT    count(DISTINCT(crashes.crash_id, units.unit_nbr, people.prsn_nbr)) AS fatalities_scooter
                               FROM      atd_txdot_crashes crashes
                               LEFT JOIN atd_txdot_units units
                               ON        (
                                                   crashes.crash_id = units.crash_id)
                               LEFT JOIN all_people people
                               ON        (
                                                   people.crash_id = crashes.crash_id
                                         AND       people.unit_nbr = units.unit_nbr)
                               WHERE     true
                               AND       crashes.crash_date >= '2014-07-28'
                               AND       crashes.crash_date <= '2023-07-28'
                               AND       units.unit_desc_id = 177
                               AND       units.veh_body_styl_id = 177
                               AND       people.prsn_injry_sev_id = 4
                               AND       crashes.in_austin_full_purpose IS true),
                     (
                               SELECT    count(DISTINCT(crashes.crash_id, units.unit_nbr, people.prsn_nbr)) AS fatalities_other
                               FROM      atd_txdot_crashes crashes
                               LEFT JOIN atd_txdot_units units
                               ON        (
                                                   crashes.crash_id = units.crash_id)
                               LEFT JOIN all_people people
                               ON        (
                                                   people.crash_id = crashes.crash_id
                                         AND       people.unit_nbr = units.unit_nbr)
                               WHERE     true
                               AND       crashes.crash_date >= '2014-07-28'
                               AND       crashes.crash_date <= '2023-07-28'
                               AND       units.unit_desc_id NOT IN (1,3,4)
                               AND       NOT (
                                                   unit_desc_id = 177
                                         AND       veh_body_styl_id = 177 )
                               AND       people.prsn_injry_sev_id = 4
                               AND       crashes.in_austin_full_purpose IS true),
                     (
                            SELECT count(DISTINCT crashes.crash_id) AS crashes_total
                            FROM   atd_txdot_crashes crashes
                            JOIN   atd_txdot_units units
                            ON     (
                                          crashes.crash_id = units.crash_id)
                            WHERE  true
                            AND    crash_date >= '2014-07-28'
                            AND    crash_date <= '2023-07-28'
                            AND    crashes.private_dr_fl != 'Y'
                            AND    crashes.in_austin_full_purpose IS true ),
                     (
                            SELECT count(DISTINCT crashes.crash_id) AS crashes_motor_vehicle
                            FROM   atd_txdot_crashes crashes
                            JOIN   atd_txdot_units units
                            ON     (
                                          crashes.crash_id = units.crash_id)
                            WHERE  true
                            AND    crash_date >= '2014-07-28'
                            AND    crash_date <= '2023-07-28'
                            AND    crashes.private_dr_fl != 'Y'
                            AND    crashes.in_austin_full_purpose IS true
                            AND    units.unit_desc_id = 1
                            AND    units.veh_body_styl_id NOT IN (71,90) ),
                     (
                            SELECT count(DISTINCT crashes.crash_id) AS crashes_motorcycle
                            FROM   atd_txdot_crashes crashes
                            JOIN   atd_txdot_units units
                            ON     (
                                          crashes.crash_id = units.crash_id)
                            WHERE  true
                            AND    crash_date >= '2014-07-28'
                            AND    crash_date <= '2023-07-28'
                            AND    crashes.private_dr_fl != 'Y'
                            AND    crashes.in_austin_full_purpose IS true
                            AND    units.unit_desc_id = 1
                            AND    units.veh_body_styl_id IN (71,90) ),
                     (
                            SELECT count(DISTINCT crashes.crash_id) AS crashes_cyclist
                            FROM   atd_txdot_crashes crashes
                            JOIN   atd_txdot_units units
                            ON     (
                                          crashes.crash_id = units.crash_id)
                            WHERE  true
                            AND    crash_date >= '2014-07-28'
                            AND    crash_date <= '2023-07-28'
                            AND    crashes.private_dr_fl != 'Y'
                            AND    crashes.in_austin_full_purpose IS true
                            AND    units.unit_desc_id = 3 ),
                     (
                            SELECT count(DISTINCT crashes.crash_id) AS crashes_pedestrian
                            FROM   atd_txdot_crashes crashes
                            JOIN   atd_txdot_units units
                            ON     (
                                          crashes.crash_id = units.crash_id)
                            WHERE  true
                            AND    crash_date >= '2014-07-28'
                            AND    crash_date <= '2023-07-28'
                            AND    crashes.private_dr_fl != 'Y'
                            AND    crashes.in_austin_full_purpose IS true
                            AND    units.unit_desc_id = 4 ),
                     (
                            SELECT count(DISTINCT crashes.crash_id) AS crashes_scooter
                            FROM   atd_txdot_crashes crashes
                            JOIN   atd_txdot_units units
                            ON     (
                                          crashes.crash_id = units.crash_id)
                            WHERE  true
                            AND    crash_date >= '2014-07-28'
                            AND    crash_date <= '2023-07-28'
                            AND    crashes.private_dr_fl != 'Y'
                            AND    crashes.in_austin_full_purpose IS true
                            AND    units.unit_desc_id = 177
                            AND    units.veh_body_styl_id = 177 ),
                     (
                            SELECT count(DISTINCT crashes.crash_id) AS crashes_other
                            FROM   atd_txdot_crashes crashes
                            JOIN   atd_txdot_units units
                            ON     (
                                          crashes.crash_id = units.crash_id)
                            WHERE  true
                            AND    crash_date >= '2014-07-28'
                            AND    crash_date <= '2023-07-28'
                            AND    crashes.private_dr_fl != 'Y'
                            AND    crashes.in_austin_full_purpose IS true
                            AND    units.unit_desc_id NOT IN (1,3,4)
                            AND    NOT (
                                          unit_desc_id = 177
                                   AND    veh_body_styl_id = 177 )),
                     (
                            SELECT count(DISTINCT units.unit_id) AS units_total
                            FROM   atd_txdot_crashes crashes
                            JOIN   atd_txdot_units units
                            ON     (
                                          crashes.crash_id = units.crash_id)
                            WHERE  true
                            AND    crash_date >= '2014-07-28'
                            AND    crash_date <= '2023-07-28'
                            AND    crashes.private_dr_fl != 'Y'
                            AND    crashes.in_austin_full_purpose IS true ),
                     (
                            SELECT count(DISTINCT units.unit_id) AS units_motor_vehicle
                            FROM   atd_txdot_crashes crashes
                            JOIN   atd_txdot_units units
                            ON     (
                                          crashes.crash_id = units.crash_id)
                            WHERE  true
                            AND    crash_date >= '2014-07-28'
                            AND    crash_date <= '2023-07-28'
                            AND    crashes.private_dr_fl != 'Y'
                            AND    crashes.in_austin_full_purpose IS true
                            AND    units.unit_desc_id = 1
                            AND    units.veh_body_styl_id NOT IN (71,90) ),
                     (
                            SELECT count(DISTINCT units.unit_id) AS units_motorcycle
                            FROM   atd_txdot_crashes crashes
                            JOIN   atd_txdot_units units
                            ON     (
                                          crashes.crash_id = units.crash_id)
                            WHERE  true
                            AND    crash_date >= '2014-07-28'
                            AND    crash_date <= '2023-07-28'
                            AND    crashes.private_dr_fl != 'Y'
                            AND    crashes.in_austin_full_purpose IS true
                            AND    units.unit_desc_id = 1
                            AND    units.veh_body_styl_id IN (71,90) ),
                     (
                            SELECT count(DISTINCT units.unit_id) AS units_cyclist
                            FROM   atd_txdot_crashes crashes
                            JOIN   atd_txdot_units units
                            ON     (
                                          crashes.crash_id = units.crash_id)
                            WHERE  true
                            AND    crash_date >= '2014-07-28'
                            AND    crash_date <= '2023-07-28'
                            AND    crashes.private_dr_fl != 'Y'
                            AND    crashes.in_austin_full_purpose IS true
                            AND    units.unit_desc_id = 3 ),
                     (
                            SELECT count(DISTINCT units.unit_id) AS units_ped
                            FROM   atd_txdot_crashes crashes
                            JOIN   atd_txdot_units units
                            ON     (
                                          crashes.crash_id = units.crash_id)
                            WHERE  true
                            AND    crash_date >= '2014-07-28'
                            AND    crash_date <= '2023-07-28'
                            AND    crashes.private_dr_fl != 'Y'
                            AND    crashes.in_austin_full_purpose IS true
                            AND    units.unit_desc_id = 4 ),
                     (
                            SELECT count(DISTINCT units.unit_id) AS units_scooter
                            FROM   atd_txdot_crashes crashes
                            JOIN   atd_txdot_units units
                            ON     (
                                          crashes.crash_id = units.crash_id)
                            WHERE  true
                            AND    crash_date >= '2014-07-28'
                            AND    crash_date <= '2023-07-28'
                            AND    crashes.private_dr_fl != 'Y'
                            AND    crashes.in_austin_full_purpose IS true
                            AND    units.unit_desc_id = 177
                            AND    units.veh_body_styl_id = 177 ),
                     (
                            SELECT count(DISTINCT units.unit_id) AS units_other
                            FROM   atd_txdot_crashes crashes
                            JOIN   atd_txdot_units units
                            ON     (
                                          crashes.crash_id = units.crash_id)
                            WHERE  true
                            AND    crash_date >= '2014-07-28'
                            AND    crash_date <= '2023-07-28'
                            AND    crashes.private_dr_fl != 'Y'
                            AND    crashes.in_austin_full_purpose IS true
                            AND    units.unit_desc_id NOT IN (1,3,4)
                            AND    NOT (
                                          unit_desc_id = 177
                                   AND    veh_body_styl_id = 177 ))) AS counts
```

---
#### Ship list
- [ ] Code reviewed
- [ ] Product manager approved

### CODE REVIEW

**Review by patrickm02L — APPROVED** [review]
## Review
Approved 💯 
Thanks again for the SQL query and thorough notes! 

## Testing
Reviewed the toggles and counts for Fatalities PR #1273 which worked perfectly. Tested all the toggles in advanced search for Crashes, and they worked like expected. 

Thanks for pointing out the 1 count difference and why that was happening - yes, I would love to talk about how to resolve it as a team.

Screen Shot 2023-08-28 at 1 06 57 PM

Screen Shot 2023-08-28 at 1 06 20 PM

File: atd-vze/src/views/Crashes/crashGridTableParameters.js [hunk 1]
```diff
@@ -354,6 +354,31 @@ export const crashGridTableAdvancedFilters = {
           ],
         },
       },
+      {
+        id: "other",
```
Comment (mddilley) [inline atd-vze/src/views/Crashes/crashGridTableParameters.js:358]: Nice! Very cool to see how these compare this to the queries in #1273. 🚀

**Review by mddilley — APPROVED** [review]
I tested this locally and also compared the totals in the UI to the DB query. Amazing to see this come down to a 1 crash difference! I'm also ready to discuss how to help the VZ team make adjustments as needed. Eager to hear more. 🚢 🙌

**Review by roseeichelmann — APPROVED** [review]
test steps worked! also ready to talk about the discrepancy whenever :-)

**Pull request merged**